pyrobosim/pyrobosim/navigation/rrt.py

```diff
@@ -261,22 +276,20 @@ def rewire_node(self, graph, n_tgt):
             dist = n.pose.get_linear_distance(n_tgt.pose)
             if (n != n_tgt) and (dist <= self.rewire_radius):
```

Comment:
Comparison here

Reply:
Based on e.g. [this](https://www.hylkerozema.nl/2021/05/10/object-oriented-programming-how-to-compare-arguments-of-methods/), the default python behavior is to compare object references. So this will return true when the variables point to the same Node object.

Overloading __eq__ would make it, e.g., so that two different Node objects with similar poses would also come out as equal, which may be helpful, but not necessary in this case?

Reply:
> which may be helpful, but not necessary in this case?

Looks like this is enough.
The issue is with the new `Graph` implementation. Will see if I can change it to use the default python mechanism instead.